announcement/announcement-tool/tool/src/java/org/sakaiproject/announcement/tool/AnnouncementAction.java

```diff
@@ -694,11 +710,38 @@ else if (view.equals(VIEW_MODE_PUBLIC))
 							{
 								messages = getMessagesPublic(site, channel, null, true, state, portlet);
 							}
+							else if (VIEW_MODE_BYROLE.equals(view))
+							{
+								messages = getMessagesByRoles(site, channel, null, true, state, portlet);
+							}
 						}
 						else
 						{
 							messages = getMessages(channel, null, true, state, portlet);
 						}
+						List<AnnouncementWrapper> finalMessages = new ArrayList<>();
+						for (AnnouncementWrapper message : messages) {
+							site = SiteService.getSite(EntityManager.newReference(message.getReference()).getContext());
+							Collection selectedRoles = message.getProperties().getPropertyList(SELECTED_ROLES_PROPERTY);
+							boolean showAnnounMessage = false;
+							if (selectedRoles == null || m_securityService.isSuperUser()) {
```

Comment:
Same as above, do it in one if.

boolean isOwner = (userDirectoryService.getCurrentUser().getId().equals(message.getAnnouncementHeader().getFrom().getId());

if (isOwner ||isAdmin || selectedRoles == null